pkg/server/api.go

```diff
@@ -136,6 +148,72 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// translateConfigForIgnitionMiddleware outputs the Ignition config in a spec version
+// understandable for the useragent and indicates whether or not the config has been
+// translated
+func translateConfigForIgnitionMiddleware(rawConf *runtime.RawExtension, useragent string) ([]byte, bool, error) {
+	translated := false
+	ignVersionString := strings.SplitAfter(useragent, "/")[1]
+	ignSemver, err := semver.NewVersion(ignVersionString)
+	if err != nil {
+		return nil, translated, errors.Errorf("invalid version in useragent: %s", useragent)
+	}
+
+	v1 := semver.New("1.0.0")
+	v2 := semver.New("2.0.0")
+	v3 := semver.New("3.0.0")
+	var reqCfgVersion semver.Version
+	if ignSemver.LessThan(*v1) {
+		// Ignition v0.x supprts config spec v2.x
+		reqCfgVersion = *v2
+	} else if !ignSemver.LessThan(*v2) && ignSemver.LessThan(*v3) {
```

Comment:
minor: since there is no such thing as ignition 1.x I suppose we can just check `ignSemver.LessThan(*v3)`

Reply:
I have no real preference here, but I though it would be best to be as concise as possible

Reply:
If there's no ignition 1, i think @yuqi-zhang suggestion is less confusing (I was confused).

Reply:
simplified the conditional here